Use Gregorian calendar regardless of the default locale

imotov opened the pull request:
Date parsing logic assumes that the calendar type is Gregorian, so it should use the Gregorian calendar even if the default locale says otherwise.

Fixes #675

imotov committed:
Use Gregorian calendar regardless of the default locale

Date parsing logic assumes that the calendar type is Gregorian, so it should use the Gregorian calendar even if the default locale says otherwise.

Fixes #675

grandinj commented on h2/src/main/org/h2/util/DateTimeUtils.java:
```diff
@@ -113,13 +113,32 @@ private static Calendar getCalendar() {
 
+    /**
+     * Creates a calendar for the default timezone.
+     *
```
please mention in the comment for this method why this is necessary.

Also please rename to createGregorianCalendar.

grandinj commented:
Nice work! thanks!

can I get a license statement, please, preferably to the mailing list, so I have a permanent record of it:

http://h2database.com/html/build.html#providing_patches

imotov committed:
Rename DateTimeUtils' createCalendar() to createGregorianCalendar()

Renames DateTimeUtils' createCalendar() to createGregorianCalendar() and improves the javadoc.

imotov commented:
Thank you for the quick review. I pushed another commit to address your comments. I also posted the license statement on the mailing list, but I don't see it there. I assume it awaits the moderator approval.
